cmd/simulator/load/loader.go

```diff
@@ -124,10 +157,61 @@ func ExecuteLoader(ctx context.Context, config config.Config) error {
 		})
 	}
 
+	go startMetricsServer(ctx, reg)
+
 	log.Info("Waiting for tx agents...")
 	if err := eg.Wait(); err != nil {
 		return err
 	}
 	log.Info("Tx agents completed successfully.")
+
+	getOutputFromMetricsServer()
 	return nil
 }
+
+func startMetricsServer(ctx context.Context, reg *prometheus.Registry) {
+	// Create a prometheus server to expose individual tx metrics
+	server := &http.Server{
+		Addr: MetricsPort,
+	}
+
+	// Start up go routine to listen for SIGINT notifications to gracefully shut down server
+	go func() {
+		defer func() {
+			if err := server.Shutdown(ctx); err != nil {
+				log.Error("Metrics server error: %v", err)
+			}
+			log.Info("Received a SIGINT signal: Gracefully shutting down metrics server")
+		}()
+
+		// Blocks until signal is received
+		<-ctx.Done()
+	}()
+
+	// Start metrics server
+	http.Handle("/metrics", promhttp.HandlerFor(reg, promhttp.HandlerOpts{Registry: reg}))
+	log.Info(fmt.Sprintf("Metrics Server: localhost%s/metrics", MetricsPort))
+	if err := server.ListenAndServe(); !errors.Is(err, http.ErrServerClosed) {
+		log.Error("Metrics server error: %v", err)
+	}
+}
+
+func getOutputFromMetricsServer() {
```

Comment:
can we return the result instead of printing it here?

Reply:
I see. in that case can we change the function name to `printOutput...`.  ? 
Thanks!

Reply:
thanks